src/timesync/meson.build

```diff
@@ -6,6 +6,14 @@ systemd_timesyncd_sources = files('''
         timesyncd-bus.h
         timesyncd-conf.c
         timesyncd-conf.h
+        timesyncd-nts-client.c
+        timesyncd-nts-client.h
+        timesyncd-ntp-extension.c
+        timesyncd-ntp-extension.h
+        timesyncd-ntske-client.c
+        timesyncd-ntske-client.h
+        timesyncd-ntske-protocol.c
+        timesyncd-ntske-protocol.h
```

Comment:
I would suggest to move all the new files in an optional meson include - that way you can remove all the \#ifdef from those sources, since there's really a lot of them, and use \#ifdef only in the common source files. Code will be much more readable and more robust that way.

Something like this should do the trick:

```
--- a/src/timesync/meson.build
+++ b/src/timesync/meson.build
@@ -13,6 +13,15 @@ systemd_timesyncd_sources = files('''
         timesyncd-server.h
 '''.split())
 
+if conf.get('GNUTLS_FOR_NTS') == 1
+        systemd_timesyncd_sources += files(['src/timesync/timesyncd-nts-client.c',
+                                            'src/timesync/timesyncd-nts-client.h',
+                                            'src/timesync/timesyncd-ntske-client.c',
+                                            'src/timesync/timesyncd-ntske-client.h',
+                                            'src/timesync/timesyncd-ntske-protocol.h',
+                                            'src/timesync/timesyncd-ntske-protocol.h'])
+endif
+
 timesyncd_gperf_c = custom_target(
         'timesyncd-gperf.c',
         input : 'timesyncd-gperf.gperf',
```

Reply:
Yeah thanks for the review.